### PR TITLE
Add visualization settings to Object detail view

### PR DESCRIPTION
**Pull request by akiselev**
Adapts the `Table` Visualization settings to `ObjectDetail` Visualization.

### CODE REVIEW

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
 
+import {
+  isMetric,
+  isDimension,
+  isNumber,
+  isURL,
+  isEmail,
+  isImageURL,
+  isAvatarURL,
+} from "metabase/lib/schema_metadata";
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx:29]: There's an import from `schema_metadata` above, we should combine them

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx [hunk 2]
```diff
@@ -456,12 +475,153 @@ export const ObjectDetailProperties = {
...
     ...columnSettings({ hidden: true }),
+    // NOTE: table column settings may be identified by fieldRef (possible not normalized) or column name:
+    //   { name: "COLUMN_NAME", enabled: true }
+    //   { fieldRef: ["field", 2, {"source-field": 1}], enabled: true }
+    "detail.columns": {
+      section: t`Columns`,
+      title: t`Visible columns`,
+      widget: ChartSettingOrderedColumns,
+      getHidden: () => false,
+      isValid: ([{ card, data }]: any) =>
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx:486]: The proper type here would be [`Series` (link)](https://github.com/metabase/metabase/blob/f5d85e8131903976d4b4194f7f5de728b522fe9c/frontend/src/metabase-types/types/Visualization.ts#L82)

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailsTable.tsx [hunk 3]
```diff
@@ -17,34 +18,58 @@ import {
   GridContainer,
   GridCell,
 } from "./ObjectDetail.styled";
+import ExternalLink from "metabase/core/components/ExternalLink";
 
 export interface DetailsTableCellProps {
   column: any;
   value: any;
+  row: any;
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailsTable.tsx:26]: The proper type here would be [`Row` (link)](https://github.com/metabase/metabase/blob/f5d85e8131903976d4b4194f7f5de728b522fe9c/frontend/src/metabase-types/types/Dataset.ts#L39)

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailsTable.tsx [hunk 4]
```diff
@@ -17,34 +18,58 @@ import {
...
   isColumnName: boolean;
   settings: any;
   className?: string;
+  getExtraDataForClick?: (clicked: any) => any;
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailsTable.tsx:30]: The proper type for `clicked` is [`ClickObject` (link)](https://github.com/metabase/metabase/blob/f5d85e8131903976d4b4194f7f5de728b522fe9c/frontend/src/metabase-types/types/Visualization.ts#L31)

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailsTable.tsx [hunk 5]
```diff
@@ -115,8 +140,23 @@ export function DetailsTable({
...
+      )
+      .filter(
+        (columnIndex: number) => columnIndex >= 0 && columnIndex < cols.length,
+      );
+
+    return {
+      cols: columnIndexes.map((i: number) => cols[i]) as any[],
+      row: columnIndexes.map((i: number) => zoomedRow[i]),
+    };
+  }, [data.cols, zoomedRow, settings]);
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailsTable.tsx:159]: Eslint complaints we need the whole `data` in the dependencies array. Can we do something to get rid of the warning, but without disabling our dep array check completely?

File: frontend/src/metabase/visualizations/components/settings/ChartSettingsTableFormatting.jsx [hunk 6]
```diff
@@ -381,11 +392,15 @@ const RuleEditor = ({ rule, cols, isNew, onChange, onDone, onRemove }) => {
...
-          />
+          {isDetail ? null : (
+            <>
+              <h3 className="mt3 mb1">{t`Highlight the whole row`}</h3>
+              <Toggle
+                value={rule.highlight_row}
+                onChange={highlight_row => onChange({ ...rule, highlight_row })}
+              />
+            </>
+          )}
```
Comment (kulyk) [inline frontend/src/metabase/visualizations/components/settings/ChartSettingsTableFormatting.jsx:403]: `!isDetail` maybe?

**Review by kulyk — APPROVED** [review]
Looks good! A few linter warnings to be fixed and it should be good to go

**Comment by iethree** [comment]
Found 2 issues here:

1) Conditional formatting doesn't work for object detail. not sure it even makes sense for object detail. 🤔 

noConditionalFormatting

2) I don't think this is an issue with object detail settings, but object detail viz in dashboards altogether: if you place it in a non-app dashboard it hits this error and creates an infinitely crashing dashboard.

Screen Shot 2022-08-17 at 10 58 53 AM

**Pull request closed without merging**